### PR TITLE
New metric: `Rand Score` 

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes part of #2003 

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
:books: Documentation preview :books:: https://torchmetrics--2025.org.readthedocs.build/en/2025/

### CODE REVIEW

File: tests/unittests/clustering/test_rand_score.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    """Check that metric rejects continuous-valued inputs."""
+    preds, target = _float_inputs
+    with pytest.raises(ValueError, match=r"Expected *"):
+        rand_score(preds, target)
+
+
+@pytest.mark.parametrize(
+    ("preds", "target"),
+    [
+        (_single_target_inputs1.preds, _single_target_inputs1.target),
```
Comment (Borda) [inline tests/unittests/clustering/test_rand_score.py:90]:
```suggestion
        (_single_target_inputs1.preds, _single_target_inputs1.target)
```

**Review by Borda — APPROVED** [review]
feeling confused seeing @matsumotosan commits in this PR, so who is the author? :)

**Comment by SkafteNicki** [comment]
> feeling confused seeing @matsumotosan commits in this PR, so who is the author? :)

I am the author but I did base this PR from the other one, so it basically just need a rebase.

File: src/torchmetrics/clustering/rand_score.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+from torchmetrics.utilities.data import dim_zero_cat
+from torchmetrics.utilities.imports import _MATPLOTLIB_AVAILABLE
+from torchmetrics.utilities.plot import _AX_TYPE, _PLOT_OUT_TYPE
+
+if not _MATPLOTLIB_AVAILABLE:
+    __doctest_skip__ = ["RandScore.plot"]
+
+
+class RandScore(Metric):
+    r"""Compute `Rand Score`_ (alternative know as Rand Index).
```
Comment (stancld) [inline src/torchmetrics/clustering/rand_score.py:29]:
```suggestion
    r"""Compute `Rand Score`_ (alternatively known as Rand Index).
```

File: src/torchmetrics/clustering/rand_score.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        self.add_state("preds", default=[], dist_reduce_fx="cat")
+        self.add_state("target", default=[], dist_reduce_fx="cat")
+
+    def update(self, preds: Tensor, target: Tensor) -> None:
+        """Update state with predictions and targets."""
+        self.preds.append(preds)
+        self.target.append(target)
+
+    def compute(self) -> Tensor:
+        """Compute mutual information over state."""
```
Comment (stancld) [inline src/torchmetrics/clustering/rand_score.py:82]:
```suggestion
        """Compute rand score over state."""
```

**Review by stancld — APPROVED** [review]

**Pull request merged**